### ISSUE TITLE
exim_queue is always 0 even if exim -bpc tells otherwise

### ISSUE THREAD

**Issue by adimenna-ripe** [gvengel/exim_exporter#63]
Dear @gvengel 

We are using the exporter on our exim servers and I noticed that the value of the exim_queue is always 0 and does not correspond to the value of `exim -bpc`

```
# exim -bpc
91
```

```
# curl http://localhost:9636/metrics 2>&1 |grep exim_queue
# HELP exim_queue Number of messages currently in queue
# TYPE exim_queue gauge
exim_queue 0
```

We are running the exporter with the following options:

```
EXIM_EXPORTER_OPTS='--web.listen-address="127.0.0.1:9636" \
--exim.log-path="/var/log/exim/" \
--exim.mainlog="main.log" \
--exim.rejectlog="reject.log" \
--exim.paniclog="panic.log" \
--exim.executable="exim" \
--exim.input-path="/var/spool/exim/input/"'
```

The rest of the metrics are correct. The input path is correct as well and does have content in there.
I enabled debug but I don't see any evident issue regrading the queue count:

```Jul 01 15:08:42 <redacted> exim_exporter[1134521]: ts=2024-07-01T15:08:42.887Z caller=main.go:228 level=debug msg="Reading queue size"```

Do you have any suggestion on how can I troubleshoot further?

Thank you

[edit]
```
# cat /etc/*release*
Oracle Linux Server release 8.9
NAME="Oracle Linux Server"
VERSION="8.9"
ID="ol"
ID_LIKE="fedora"
VARIANT="Server"
VARIANT_ID="server"
VERSION_ID="8.9"
PLATFORM_ID="platform:el8"
PRETTY_NAME="Oracle Linux Server 8.9"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:oracle:linux:8:9:server"
HOME_URL="https://linux.oracle.com/"
BUG_REPORT_URL="https://github.com/oracle/oracle-linux"
```

**Comment by Chronos85** [comment]
So i had the same problem and debuggen it a bit. In my case, and i suspect its the same with you, exim has updated its MessageID specs:

https://lists.exim.org/lurker/message/20231104.135832.37148bbd.en.html 

The code checks if the filename length is 18, but that no longer is valid on newer releases
```
diff --git a/main.go b/main.go
index 10ea9d2..a2afa55 100644
--- a/main.go
+++ b/main.go
@@ -219,7 +219,7 @@ func (e *Exporter) CountMessages(dirname string) float64 {
        }
        var count float64
        for _, name := range messages {
-               if len(name) == 18 && strings.HasSuffix(name, "-H") {
+               if ( len(name) == 18 || len(name) == 25) && strings.HasSuffix(name, "-H") {
                        count += 1
                }
        }
```
This actually fixes it

**Comment by gvengel** [comment]
Thanks for the report, @adimenna-ripe. Yes as @Chronos85 mentioned, there a couple compatibility issues with exim 4.97. I have time set aside this week to get them sorted out, along with a few other things, then I'll cut a new release.

**Comment by gvengel** [comment]
Fixed in 0.6.2.
